lib/Notification/Notifier.php

```diff
@@ -445,18 +445,15 @@ protected function parseCall(INotification $notification, Room $room, IL10N $l):
 	 * @return INotification
 	 * @throws \InvalidArgumentException
 	 */
-	protected function parsePasswordRequest(INotification $notification, Room $room, IL10N $l) {
-		if ($notification->getObjectType() !== 'room') {
+	protected function parsePasswordRequest(INotification $notification, Room $room, IL10N $l): INotification {
+		if ($notification->getObjectType() !== 'call') {
 			throw new \InvalidArgumentException('Unknown object type');
 		}
 
-		$parameters = $notification->getSubjectParameters();
-		$sharedWith = $parameters['sharedWith'];
+		$sharedWith = $room->getName();
```

Comment:
I am not a fan of getting the e-mail address from the room name; I would use a parameter instead.

Right now it is OK because the Talk sidebar is shown only for mail shares, but in the (very very near ;-) ) future it will be available for link shares too, so it will be necessary to know the share type too to set the proper message (which will probably need the file name of the share too).

Reply:
Ah, well this is hte normal call notification, so no new parameters.
But I guess I can get the share based on the object info and from there get the info we are looking for.

Reply:
I pushed a commit with what I think you meant; please check if I got it right ;-)